tez-dag/src/main/java/org/apache/tez/dag/app/dag/impl/VertexImpl.java

```diff
@@ -2817,6 +2817,18 @@ private VertexState setupVertex() {
     return VertexState.INITED;
   }
 
+  private boolean canSkipInitializingParents() {
+    // Both cases use RootInputVertexManager. RootInputVertexManager can start tasks even though
+    // any parents are not fully initialized.
+    if (vertexPlan.hasVertexManagerPlugin()) {
+      final VertexManagerPluginDescriptor pluginDesc = DagTypeConverters
+          .convertVertexManagerPluginDescriptorFromDAGPlan(vertexPlan.getVertexManagerPlugin());
+      return pluginDesc.getClassName().equals(RootInputVertexManager.class.getName());
+    } else {
+      return inputsWithInitializers != null;
```

Comment:
in the else branch: it's not a RootInputVertexManager, but has inputsWithInitializers, what kind of plugin is used here if any?

Reply:
Based on the comment above, I guess it would try to care about this pattern. Honestly, I lost a part of my memory about this PR.
https://github.com/apache/tez/blob/ca1511998e183c8e10da79bef20463d40a39726a/tez-dag/src/main/java/org/apache/tez/dag/app/dag/impl/VertexImpl.java#L2898-L2917

Reply:
okay, that makes sense, are you fine with further refactoring and reusing the conditions where possible? something like below:

```
private boolean isRootInputVertexManagerUsed() {
  return isRootInputVertexManagerPluginUsed() || hasInputInitializers();
}

private boolean isRootInputVertexManagerPluginUsed(){
return vertexPlan.hasVertexManagerPlugin() && DagTypeConverters
       .convertVertexManagerPluginDescriptorFromDAGPlan(vertexPlan.getVertexManagerPlugin()).getClassName().equals(RootInputVertexManager.class.getName());
}

private boolean hasInputInitializers(){
  return inputsWithInitializers != null;
}
```
hasInputInitializers might be used in 2 places for cross reference

Reply:
Done.
https://github.com/apache/tez/pull/361/commits/c8c18fcb49de1d0cadd368ed8db73750f9d154f7

I kept this branch intentionally because there might be a case where a vertex manager plugin is explicitly configured but it is not a RootInputVertexManager.

```
    if (vertexPlan.hasVertexManagerPlugin()) {
      final VertexManagerPluginDescriptor pluginDesc = DagTypeConverters
          .convertVertexManagerPluginDescriptorFromDAGPlan(vertexPlan.getVertexManagerPlugin());
      return pluginDesc.getClassName().equals(RootInputVertexManager.class.getName());
    } else {
      // This case implicitly uses RootInputVertexManager. See VertexImpl#assignVertexManager
      return hasInputInitializers();
    }
```